site/app/views/forum/ForumThreadView.php

```diff
@@ -619,6 +619,11 @@ public function displayThreadList($threads, $filtering, &$activeThreadAnnounceme
             if ($thread["deleted"]) {
                 $class .= " deleted";
             }
+
+            if ($this->core->getQueries()->getUserById($thread['created_by'])->getGroup() <= 3) {
```

Comment:
It is preferred to use constants instead of random numbers for readability.

```suggestion
            if ($this->core->getQueries()->getUserById($thread['created_by'])->getGroup() <= User::GROUP_LIMITED_ACCESS_GRADER) {
```

Reply:
Also to conform with MVC architecture, it is recommended not to get queries in your view function. Instead, it is better to put the queries in controllers and pass them to the view.

It is an ongoing task to forbid people from doing queries in views.

https://github.com/Submitty/Submitty/issues/4537

Reply:
Thanks for catching this @zjxiaohan ! It looks like there are a lot of other queries also being made in this file as well there.

 It might be worth merging this PR for now and making a separate PR to move all of them to the forum controller at once but I'm not too familiar with this area @andrewaikens87 @elihschiff  so not sure whats easier/makes more sense.

Reply:
@shailpatels Thanks for the reply. I didn't notice that there are a lot of queries in this file... Doing it in separate PR(s) would be better as far as I am concerned.

Reply:
I actually think I found an even better solution. I switched it to use the accessGrading() function which does exactly what I was doing before but is a member function of user so there are no random numbers.

Reply:
@elihschiff Yeah true that's the best solution. I haven't committing code for a long time and forgot that we have `accessGrading`. :)